workflow/validate/validate.go

```diff
@@ -243,6 +235,15 @@ func (ctx *templateValidationCtx) validateTemplate(tmpl *wfv1.Template, tmplCtx
 	if err != nil {
 		return errors.Errorf(errors.CodeBadRequest, "templates.%s %s", tmpl.Name, err)
 	}
+
```

Comment:
Move this part here from the beginning to make it runs `common.ProcessArgs` every time.

This is because `ProcessArgs` is not invoked any more in `ResolveTemplate` (function in `context.go`, called by `validateTemplateHolder`) with the change #1744.